web/test.html

```diff
@@ -1,110 +1,210 @@
-  }
-
-  function doConnect()
-  {
-    websocket = new WebSocket(document.myform.url.value);
-    websocket.onopen = function(evt) { onOpen(evt) };
-    websocket.onclose = function(evt) { onClose(evt) };
-    websocket.onmessage = function(evt) { onMessage(evt) };
-    websocket.onerror = function(evt) { onError(evt) };
-  }
-
-  function onOpen(evt)
-  {
-    writeToScreen("connected\n");
-	document.myform.connectButton.disabled = true;
-	document.myform.disconnectButton.disabled = false;
-  }
-
-  function onClose(evt)
-  {
-    writeToScreen("disconnected\n");
-	document.myform.connectButton.disabled = false;
-	document.myform.disconnectButton.disabled = true;
-  }
-
-  function onMessage(evt)
-  {
-    writeToScreen("response: " + evt.data + '\n');
-  }
-
-  function onError(evt)
-  {
-    writeToScreen('error: ' + evt.data + '\n');
-
-	websocket.close();
-
-	document.myform.connectButton.disabled = false;
-	document.myform.disconnectButton.disabled = true;
-
-  }
-
-  function doSend(message)
-  {
-    writeToScreen("sent: " + message + '\n'); 
-    websocket.send(message);
-  }
-
-  function writeToScreen(message)
-  {
-    document.myform.outputtext.value += message
-	document.myform.outputtext.scrollTop = document.myform.outputtext.scrollHeight;
-
-  }
-
-  window.addEventListener("load", init, false);
-
-
-   function sendText() {
-		doSend( document.myform.inputtext.value );
-   }
-
-  function clearText() {
-		document.myform.outputtext.value = "";
-   }
-
-   function doDisconnect() {
-		websocket.close();
-   }
-
-
-</script>
-
-<div id="output"></div>
-
-<form name="myform">
-<p>
-<textarea name="outputtext" rows="20" cols="50"></textarea>
-</p>
-<p>
-<textarea name="inputtext" cols="50"></textarea>
-</p>
-<p>
-<textarea name="url" cols="50"></textarea>
-</p>
-<p>
-<input type="button" name=sendButton value="Send" onClick="sendText();">
-<input type="button" name=clearButton value="Clear" onClick="clearText();">
-<input type="button" name=disconnectButton value="Disconnect" onClick="doDisconnect();">
-<input type="button" name=connectButton value="Connect" onClick="doConnect();">
-</p>
-
-
-</form>
-</html> 
-
-
+<!DOCTYPE html>
+
+<meta charset="utf-8" />
+
+<title>WebSocket Test</title>
+
+<style>
+  * {
+    box-sizing: border-box;
+  }
+
+  /* Create two equal columns that floats next to each other */
+  .column {
+    float: left;
+    width: 50%;
+    padding: 1%;
+  }
+
+  /* Clear floats after the columns */
+  .row:after {
+    content: "";
+    display: table;
+    clear: both;
+  }
+
+
+  #image {
+    max-height: 100%;
+    max-width: 48%;
+    position: absolute;
+    border: 1px solid black;
+
+  }
+</style>
+
+
+<script language="javascript" type="text/javascript">
+
+  //add event listener for page load
+  window.addEventListener("load", init, false);
+
+  //add event listener for key press
+  document.addEventListener('keyup', (e) => {
+    if (e.code === "ArrowLeft") {

```

Comment:
Can we change line 44-50 to simply forward all key events, then we can filter in the back end instead. We could also try to be consistent in the format with the mouse click events:

doSend("EVENT(key)=" + e.code)